tools/staging_migrator/src/molgenis_emx2_staging_migrator/utils.py

```diff
@@ -0,0 +1,203 @@
+"""
+Utility functions for the StagingMigrator class.
+"""
+import logging
+from io import BytesIO
+
+import pandas as pd
+
+from tools.pyclient.src.molgenis_emx2_pyclient.exceptions import IncorrectSchemaError
+from tools.staging_migrator.src.molgenis_emx2_staging_migrator.graphql_queries import Queries
+
+log = logging.getLogger(__name__)
+
+
+def get_cohort_ids(server_url, session, staging_area) -> list | None:
+    """Fetches the id associated with the staging area's cohort."""
+
+    # Query server for cohort id
+    query = Queries.Cohorts
+    staging_url = f"{server_url}/{staging_area}/graphql"
+    response = session.post(url=staging_url,
+                            json={"query": query})
+    response_data = response.json().get('data')
+    if response_data is None:
+        # Raise new error
+        raise IncorrectSchemaError(f"Table 'Cohorts' not found on schema '{staging_area}'.")
+
+    # Return only if there is exactly one id/cohort in the Cohorts table
+    if "Cohorts" in response_data.keys():
+        if len(response_data['Cohorts']) < 1:
+            log.warning(
+                f'Expected a cohort in staging area "{staging_area}"'
+                f' but found {len(response_data["Cohorts"])}')
+            return None
+    else:
+        log.warning(
+            f'Expected a single cohort in staging area "{staging_area}"'
+            f' but found none.')
+        return None
+
+    return [cohort['id'] for cohort in response_data['Cohorts']]
+
+
+def prepare_pkey(schema: dict, table_name: str, col_id: str | list = None) -> str | list | dict | None:
```

Comment:
looks like a generic function , suggest to move it into the client package ( maybe have a utils.py there with generic util functions )